src/dmd/mtype.d

```diff
@@ -2805,18 +2805,15 @@ extern (C++) abstract class Type : RootObject
         auto name = namelen <= namebuf.length ? namebuf.ptr : cast(char*)malloc(namelen);
         assert(name);
 
-        const length = sprintf(name, "_D%lluTypeInfo_%.*s6__initZ",
-                cast(ulong)(9 + slice.length), cast(int)slice.length, slice.ptr);
+        // If C mangling will add the prefix in, don't add our own.
+        const (char)* prefix = Target.cManglePrefix[0] ? "" : "_";
```

Comment:
Why not go the full chain and add `Target.customCManglePrefix()`?

Reply:
I have no idea where DMD does mangling in the backend, I'm not sure whether there's scope to re-use this field in other parts of the compiler also, maybe even expose it to the user code somehow (traits, pragma, mangleof?).

In GCC and LLVM internals expose this as a `__USER_LABEL_PREFIX__` [macro](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html).

Reply:
I'm not aware of any target defining this as anything other than `"_"` or `""`.  So it could be a `char` instead.  It's a string here though just to be compatible with GCC and LLVM.

Reply:
Right, and so does GDC for the exact same reasons.  If you have any better ideas that don't involve changing the dmd backend, then send them my way.

Reply:
Such as, removing this ABI hack altogether.

Reply:
Also, doesn't LLVM expose a baseline ABI?  Not that I'm making accusations over what they consider best practices (you may notice that I get confused by a lot of things that LLVM does).